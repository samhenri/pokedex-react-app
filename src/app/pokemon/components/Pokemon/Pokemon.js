import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Pokemon.scss';

const CN = 'pokemon';

export class Pokemon extends Component {
  static propTypes = {
    className: PropTypes.string,
    isFetching: PropTypes.bool,
    getPokemon: PropTypes.func,
    match: PropTypes.object,
    pokemon: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  };

  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {
      match: { params },
      getPokemon,
    } = this.props;

    getPokemon(params.pokemon);
  }

  componentDidUpdate() {
    // const { pokemon } = this.props;
    // if (Object.keys(pokemon).length !== 0 && pokemon.constructor === Object) {
    //   this.removeLoading();
    // }
  }

  removeLoading() {
    this.setState({
      isFetching: false,
    });
  }

  render() {
    const { className, pokemon, isFetching } = this.props;

    console.log(pokemon);

    const pokemonWrapperClassName = classNames(`${CN}__wrapper`, className);
    const pokemonClassName = classNames(`${CN}__main`);

    return (
      <div className={pokemonWrapperClassName}>
        <div className={pokemonClassName}>
          Pokemon: <br />
          evolution <br />
          species:
        </div>
        {isFetching && 'Loading... '}
      </div>
    );
  }
}
