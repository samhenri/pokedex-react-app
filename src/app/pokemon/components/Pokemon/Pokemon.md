## Pok√©mon Entity Component
    <Pokemon id="1" name="bulbasaur" type="grass" height="0.7" weight="6.9" />
