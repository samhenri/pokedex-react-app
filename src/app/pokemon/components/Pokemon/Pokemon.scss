.pokemon {
  &__wrapper {
    position: relative;

    display: flex;

    width: 375px;
    height: 812px;
    margin: 0 auto;
  }
}
