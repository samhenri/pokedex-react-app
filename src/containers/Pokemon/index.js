import React from 'react';

const Pokemon = () => (
  <div>
    <h1>About Page</h1>
    <p>Did you get here via Redux?</p>
  </div>
);

export default Pokemon;
