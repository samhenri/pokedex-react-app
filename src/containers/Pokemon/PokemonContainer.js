import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { getPokemon } from '../../actions/getPokemon';

import { Pokemon } from '../../app/pokemon/components/Pokemon';

const mapStateToProps = (state) => {
  return {
    //pokemon: { ...state.pokemon, ...state.species, ...state.evolution },
    pokemon: state.pokemon,
    isFetching: Object.keys(state.pokemon).length === 0 ? true : false,
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      getPokemon,
    },
    dispatch,
  );

export const PokemonContainer = connect(mapStateToProps, mapDispatchToProps)(
  Pokemon,
);
