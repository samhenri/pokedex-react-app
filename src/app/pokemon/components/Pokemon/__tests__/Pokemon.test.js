import React from 'react';
import { shallow } from 'enzyme';
import { Pokemon } from '../Pokemon';

describe('<Pokemon /> component', () => {
  it('should render Bulbasaur type', () => {
    const component = shallow(
      <Pokemon id="1" name="bulbasaur" type="grass" height="0.7" weight="6.9" />,
    );
    expect(component).toMatchSnapshot();
  });
});
