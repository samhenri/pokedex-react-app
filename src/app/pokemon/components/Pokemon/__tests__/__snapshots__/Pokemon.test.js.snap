// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon /> component should render Bulbasaur type 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pokemon
    height={0.7}
    id={1}
    name="Bulbasaur"
    type="grass"
    weight={6.9}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Dex entry: ",
        1,
        " ",
        <br />,
        "Name: ",
        "Bulbasaur",
        " ",
        <br />,
        "Type: ",
        "grass",
        " ",
        <br />,
        "Height: ",
        0.7,
        " ",
        <br />,
        "Weight: ",
        6.9,
        " ",
        <br />,
      ],
      "className": "pokemon",
    },
    "ref": null,
    "rendered": Array [
      "Dex entry: ",
      1,
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Name: ",
      "Bulbasaur",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Type: ",
      "grass",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Height: ",
      0.7,
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Weight: ",
      6.9,
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Dex entry: ",
          1,
          " ",
          <br />,
          "Name: ",
          "Bulbasaur",
          " ",
          <br />,
          "Type: ",
          "grass",
          " ",
          <br />,
          "Height: ",
          0.7,
          " ",
          <br />,
          "Weight: ",
          6.9,
          " ",
          <br />,
        ],
        "className": "pokemon",
      },
      "ref": null,
      "rendered": Array [
        "Dex entry: ",
        1,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Name: ",
        "Bulbasaur",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Type: ",
        "grass",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Height: ",
        0.7,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Weight: ",
        6.9,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
